pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -533,7 +605,7 @@ func (nc *NodeController) MonitorNodeStatus() error {
 		var lastReadyCondition api.NodeCondition
 		node := &nodes.Items[i]
 		for rep := 0; rep < nodeStatusUpdateRetry; rep++ {
-			err, gracePeriod, lastReadyCondition = nc.tryUpdateNodeStatus(node)
+			gracePeriod, lastReadyCondition, err = nc.tryUpdateNodeStatus(node)
 			if err != nil {
```

Comment:
orthogonal to the PR, but go prefers 'early return', so it would be better to 
if err == nil {
   break
}
name := node.Name
xxx